modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/VerticalBar.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {VerticalBar} from '@clayui/core';
+import {navigate} from 'frontend-js-web';
+import React from 'react';
+
+const CSS_EXPANDED = 'expanded';
+
+const VerticalNavigationBar = ({items, parentContainerId}) => {
+	const parentContainer = document.getElementById(parentContainerId);
+
+	const activeItem = items.find((item) => item.active);
+
+	const onIconClick = (item) => {
+		if (item.key !== activeItem.key) {
+			if (!parentContainer.classList.contains(CSS_EXPANDED)) {
+				parentContainer.classList.add(CSS_EXPANDED);
+			}
```

Comment:
I think this check is a litter redundant, we can add it without testing.

https://stackoverflow.com/questions/36999220/checking-classlist-with-contains-if-a-class-exists-before-add-or-remove

Reply:
Didn't know it, thanks for the tip!